src/agent/restart/src/restart_unix.cpp

```diff
@@ -0,0 +1,117 @@
+#include "restart_unix.hpp"
+#include <fstream>
+#include <logger.hpp>
+
+namespace restart
+{
+    std::vector<char*> GetCommandLineArgs()
+    {
+        std::vector<char*> args;
+        std::ifstream cmdline_file("/proc/self/cmdline");
+
+        if (!cmdline_file)
+        {
+            LogError("Failed to open /proc/self/cmdline.");
+            return args;
+        }
+
+        std::string arg;
+        while (getline(cmdline_file, arg, '\0'))
+        {
+            args.push_back(strdup(arg.c_str()));
+        }
+
+        args.push_back(nullptr);
+
+        return args;
+    }
+
+    bool UsingSystemctl()
+    {
+        return (0 == std::system("which systemctl > /dev/null 2>&1") && nullptr != std::getenv("INVOCATION_ID"));
+    }
+
+    boost::asio::awaitable<module_command::CommandExecutionResult> RestartWithSystemd()
+    {
+        LogInfo("Systemctl restarting wazuh agent service.");
+        if (std::system("systemctl restart wazuh-agent") != 0)
```

Comment:
Can we try to avoid system calls? Maybe there is some library to restart the systemd services without using `std::system`?

Reply:
I did some research, and it doesn't seem straightforward to avoid system calls for restarting systemd services. For now, I would prefer to leave it as is. We could consider improving it in the future. What do you think?